lib/ramble/ramble/spack_runner.py

```diff
@@ -326,14 +327,51 @@ def add_include_file(self, include_file):
         if file_name in self._allowed_config_files:
             self.includes.append(include_file)
 
-    def concretize(self):
+    def copy_from_external_env(self, env_name_or_path):
         """
-        Concretize a spack environment.
+        Copy an external spack environment file into the generated environment.
 
-        This happens by generating a spack.yaml file to build
-        the packages that should be in the environment.
+        env_name_or_path can be either:
+         - Name of a named spack environment
+         - Path to an external spack environment
 
-        This command requires an active spack environment.
+        Returns:
+         - (bool) found_lock: True if a spack.lock file was copied. False otherwise.
+        """
+
+        self._check_active()
+
+        named_location_args = [
+            'location',
+            '-e',
+            env_name_or_path
+        ]
+
+        # Try to use a named environment first
+        try:
+            path = self.exe(*named_location_args, output=str).strip('\n')
+        # If a named environment fails, copy directly from the path
+        except ProcessError:
+            path = env_name_or_path
+
+        found_lock = False
+
+        lock_file = os.path.join(path, 'spack.lock')
+        if os.path.exists(lock_file):
+            found_lock = True
+            shutil.copyfile(lock_file, os.path.join(self.env_path, 'spack.lock'))
+
+        conf_file = os.path.join(path, 'spack.yaml')
+        if not os.path.exists(conf_file):
+            raise InvalidExternalEnvironment(f'{path} is not a spack environment.')
+
+        shutil.copyfile(conf_file, os.path.join(self.env_path, 'spack.yaml'))
```

Comment:
Are you worried about clobbering existing confs?

Reply:
No, I purposefully want to clobber existing configs. Currently, ramble's semantics are that when you run `workspace setup` it regenerates the `spack.yaml` (even if it's identical). Also, it defaults to forcibly re-concretizing (clobbering a `spack.lock`).

So, I think this makes the external environments behave the same way internal ones behave.

Reply:
No, I purposefully want to clobber existing configs. Currently, ramble's semantics are that when you run `workspace setup` it regenerates the `spack.yaml` (even if it's identical). Also, it defaults to forcibly re-concretizing (clobbering a `spack.lock`).

So, I think this makes the external environments behave the same way internal ones behave.